### PR TITLE
Upgrade optimum-habana diffusers dependency from 0.26.3 to 0.29.2

### PR DESCRIPTION
**Pull request by dsocek**
# Upgrade optimum-habana diffusers from 0.26.3 to 0.29.2

* Updated optimum-habana diffusers to 0.29.2 in `setup.py`
* Fixed SVD (GaudiStableVideoDiffusionPipeline) to handle updated API in 0.29.2 for `tensor2vid` and instead use `video_processor.postprocess_video`
* Fixed GaudiStableDiffusionControlNetPipeline to handle updated `check_inputs` API in 0.29.2
* Fixed `tests/test_diffusers.py` to use `to(dtype=...)` instead of `to(torch_dtype=...)` which is deprecated starting diffusers 0.27.0 (See [here](https://github.com/huggingface/diffusers/blob/v0.26.3/src/diffusers/pipelines/pipeline_utils.py#L781))
* Clean-up SVD to remove all inheritable functions
* Updated warmup logic in SVD
* Updated documentation example with command line for SVD that yields improved performance

## Tests
Ran tests on G2:

```bash
export RUN_SLOW=true
export GAUDI2_CI=1
make test_installs
python -m pytest tests/test_diffusers.py
```

Diffusers test results **before** diffusers upgrade:
```bash
17 failed, 106 passed, 13 skipped, 65 warnings in 2304.15s (0:38:24)
```

Diffusers test results **after** diffusers upgrade:
```bash
16 failed, 107 passed, 13 skipped, 316 warnings in 2157.94s (0:35:57)
```

**Note:**  Update is needed for supporting enabling newest pipelines such as **Stable Diffusion 3**, etc.

### CODE REVIEW

File: examples/stable-diffusion/README.md [hunk 1]
```diff
@@ -445,6 +445,44 @@ python image_to_image_generation.py \
     --bf16
 ```
 
+## Inpainting Example
+
+Inpainting replaces or edits specific areas of an image. For more details,
+please refer to [Huging Face Diffusers doc](https://huggingface.co/docs/diffusers/en/using-diffusers/inpaint).
```
Comment (regisss) [inline examples/stable-diffusion/README.md:451]:
```suggestion
please refer to [Hugging Face Diffusers doc](https://huggingface.co/docs/diffusers/en/using-diffusers/inpaint).
```

File: setup.py [hunk 2]
```diff
@@ -47,6 +47,7 @@
     "datasets",
     "safetensors",
     "pytest < 8.0.0",
+    "scipy",
```
[the remark this answers is not in the document]
Comment (dsocek) [reply setup.py:50]: There is one:  `StableDiffusionInpaintPipelineFastTests`

```sh
FAILED tests/test_diffusers.py::StableDiffusionInpaintPipelineFastTests::test_karras_schedulers_shape - ImportError:
```

```sh
tests/test_diffusers.py:3168:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/usr/local/lib/python3.10/dist-packages/diffusers/utils/dummy_torch_and_scipy_objects.py:13: in from_config
    requires_backends(cls, ["torch", "scipy"])
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

obj = <class 'diffusers.utils.dummy_torch_and_scipy_objects.LMSDiscreteScheduler'>, backends = ['torch', 'scipy']

    def requires_backends(obj, backends):
        if not isinstance(backends, (list, tuple)):
            backends = [backends]

        name = obj.__name__ if hasattr(obj, "__name__") else obj.__class__.__name__
        checks = (BACKENDS_MAPPING[backend] for backend in backends)
        failed = [msg.format(name) for available, msg in checks if not available()]
        if failed:
>           raise ImportError("".join(failed))
E           ImportError:
E           LMSDiscreteScheduler requires the scipy library but it was not found in your environment. You can install it with pip: `pip install
E           scipy`

/usr/local/lib/python3.10/dist-packages/diffusers/utils/import_utils.py:611: ImportError
```

File: setup.py [hunk 3]
```diff
@@ -33,8 +33,8 @@
     "optimum",
     "torch",
     "accelerate < 0.28.0",
-    "diffusers >= 0.26.0, < 0.27.0",
-    "huggingface_hub < 0.23.0",
+    "diffusers == 0.29.2",
+    "huggingface_hub == 0.23.2",
```
[the remark this answers is not in the document]
Comment (dsocek) [reply setup.py:37]: I can make it >=

**Review by regisss — APPROVED** [review]
LGTM!
I removed the `datasets < 2.20.0` line in `setup.py`. You may have got issues with datasets that needs `trust_remote_code=True`.

**Pull request merged**